FrostyEditor/Windows/MainWindow.xaml.cs

```diff
@@ -386,19 +386,29 @@ private void launchButton_Click(object sender, RoutedEventArgs e)
             // setup ability to cancel the process
             CancellationTokenSource cancelToken = new CancellationTokenSource();
 
-            Random r = new Random();
-            string editorModName = $"EditorMod{r.Next(1000, 9999).ToString("D4")}.fbmod";
             launchButton.IsEnabled = false;
 
             // get all mods
             List<string> modPaths = new List<string>();
-
+            
             DirectoryInfo modDirectory = new DirectoryInfo($"Mods/{ProfilesLibrary.ProfileName}");
             foreach (string modPath in Directory.EnumerateFiles($"Mods/{ProfilesLibrary.ProfileName}/", "*.fbmod", SearchOption.AllDirectories))
-                modPaths.Add(Path.GetFileName(modPath));
+            {
+                if(Path.GetFileName(modPath).Contains("EditorMod"))
+                    File.Delete(modPath);
+                else
+                    modPaths.Add(Path.GetFileName(modPath));
+            }
+            
+            Random r = new Random();
+            string editorModName = $"EditorMod_{r.Next(1000, 9999).ToString("D4")}.fbmod";
+            while (File.Exists(editorModName))
```

Comment:
dont all the editor mods get deleted in the foreach above, so this is unnecessary

Reply:
> dont all the editor mods get deleted in the foreach above, so this is unnecessary

I delete that while